spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/MongoDbEnvironmentRepository.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.config.server.environment;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.mongodb.MongoException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.config.environment.Environment;
+import org.springframework.cloud.config.environment.PropertySource;
+import org.springframework.core.Ordered;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Alexandros Pappas
+ */
+public class MongoDbEnvironmentRepository implements EnvironmentRepository, Ordered {
+
+	private static final Log logger = LogFactory.getLog(JdbcEnvironmentRepository.class);
+
+	private final MongoTemplate mongoTemplate;
+
+	private final MongoDbEnvironmentProperties properties;
+
+	public MongoDbEnvironmentRepository(MongoTemplate mongoTemplate, MongoDbEnvironmentProperties properties) {
+		this.mongoTemplate = mongoTemplate;
+		this.properties = properties;
+	}
+
+	@Override
+	public Environment findOne(String application, String profile, String label) {
+		label = Optional.ofNullable(label).filter(StringUtils::hasText).orElse(this.properties.getDefaultLabel());
+		profile = Optional.ofNullable(profile).filter(StringUtils::hasText).orElse("default");
```

Comment:
We don't use Optional except for certain return and parameter situations. Please use the ternary operator